esp-wifi/src/lib.rs

```diff
@@ -6,13 +6,19 @@
 #![doc = include_str!("../README.md")]
 #![doc(html_logo_url = "https://avatars.githubusercontent.com/u/46717278")]
 
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
 // MUST be the first module
 mod fmt;
 
+#[cfg(not(feature = "alloc"))]
```

Comment:
Instead of this cfg mess, could you please introduce a conditional `alloc` module?

```rust
#[cfg(not(feature = "alloc"))]
mod alloc;

...

#[cfg(not(feature = "alloc"))]
alloc::init_heap();
```

This way hopefully we can get away with 2 `cfg`s